src/blocks/DetailsBlock/DetailsBlock.astro

```diff
@@ -0,0 +1,45 @@
+---
+import type { DetailsBlockFragment } from '@lib/types/datocms';
+
+interface Props {
+  block: DetailsBlockFragment
+}
+
+const { block } = Astro.props;
+const splitChar = '|';
+const hasMultipleValues = (value: string) => value.includes(splitChar);
+const splitValues = (value: string) => value.split(splitChar);
+---
+
+<dl class:list={['details-block', `details-block--${block.display}`]}>
+  {
+    Object.keys(block.details).map((key) => (
```

Comment:
I think you can simplify this to:

```js
Object.entries(block.details).map(([key, value]) =>
```
and later use
```js
value.split(separator).map(text => <dd>{ text }</dd>)
```

so you don't need helper functions and don't need to distinguish between single and multiple value. Splitting a single value will just return an array with a single item.

Reply:
Good one!